dev_tools/shell_tools.py

```diff
@@ -290,19 +290,19 @@ def output_of(*cmd: Optional[str], **kwargs) -> str:
 
     Args:
         *cmd: Components of the command to execute, e.g. ["echo", "dog"].
-        **kwargs: Extra arguments for asyncio.create_subprocess_shell, such as
+            Arguments set to None are disregarded in command execution.
+        **kwargs: Extra arguments for the shell_tools.run function, such as
             a cwd (current working directory) argument.
 
     Returns:
-        A (captured output, captured error output, return code) triplet. The
-        captured outputs will be None if the out or err parameters were not set
-        to an instance of TeeCapture.
+        str: The standard output of the command with the last newline removed.
 
     Raises:
          subprocess.CalledProcessError: The process returned a non-zero error
-            code and raise_on_fail was set.
+            code and the `check` flag was True (default).
     """
-    result = cast(str, run_cmd(*cmd, log_run_to_stderr=False, out=TeeCapture(), **kwargs).out)
+    cmdpure = [w for w in cmd if w is not None]
```

Comment:
nit: Should this check be inside `run()`? Do all callers of `run()` need to check this?

Reply:
I'd like to have a minimum difference between behavior of shell_tools.run and subprocess.run, so I'd rather not do any filtering of the cmd arguments.

For consistency sake it is perhaps better to have output_of() handle arguments in the same way as run() and do the argument filtering in the caller.  I have checked the references to output_of and it seems easily doable.